examples/bottom_navigation_multiple_beamers/lib/main.dart

```diff
@@ -164,94 +156,113 @@ class ArticlesLocation extends BeamLocation {
-              articleId: state.pathParameters['articleId'],
+              article: articles.firstWhere((article) =>
+                  article['id'] == state.pathParameters['articleId']),
             ),
           ),
       ];
 }
 
 // APP
 class AppScreen extends StatefulWidget {
-  AppScreen({this.beamState});
-
+  AppScreen({required this.beamState});
   final BeamState beamState;
 
   @override
   _AppScreenState createState() => _AppScreenState();
 }
 
 class _AppScreenState extends State<AppScreen> {
-  final _routerDelegates = [
+  late int currentIndex;
+
+  final routerDelegates = [
     BeamerRouterDelegate(
-      locationBuilder: (state) => ArticlesLocation(state),
+      locationBuilder: (state) => BooksLocation(state),
     ),
     BeamerRouterDelegate(
-      locationBuilder: (state) => BooksLocation(state),
+      locationBuilder: (state) => ArticlesLocation(state),
     ),
   ];
 
-  int _currentIndex;
-
   @override
   void initState() {
+    currentIndex = widget.beamState.uri.path.contains('books') ? 0 : 1;
     super.initState();
-    _currentIndex = widget.beamState.uri.path.contains('books') ? 1 : 0;
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       body: IndexedStack(
-        index: _currentIndex,
+        index: currentIndex,
         children: [
-          Beamer(routerDelegate: _routerDelegates[0]),
+          Beamer(routerDelegate: routerDelegates[0]),
           Container(
             color: Colors.blueAccent,
             padding: const EdgeInsets.all(32.0),
-            child: Beamer(routerDelegate: _routerDelegates[1]),
+            child: Beamer(routerDelegate: routerDelegates[1]),
           ),
         ],
       ),
       bottomNavigationBar: BottomNavigationBar(
-        currentIndex: _currentIndex,
+        currentIndex: currentIndex,
         items: [
-          BottomNavigationBarItem(label: 'A', icon: Icon(Icons.article)),
-          BottomNavigationBarItem(label: 'B', icon: Icon(Icons.book)),
+          BottomNavigationBarItem(label: 'Books', icon: Icon(Icons.book)),
+          BottomNavigationBarItem(label: 'Articles', icon: Icon(Icons.article)),
         ],
         onTap: (index) {
-          setState(() => _currentIndex = index);
-          _routerDelegates[_currentIndex].parent?.updateRouteInformation(
-                _routerDelegates[_currentIndex].currentLocation.state.uri,
-              );
+          setState(() => currentIndex = index);
+
+          // todo:
+
+          // OPTION 1:
+          // Advantage: URL updates
+          // Disadvantage: pops details pages and shows pop animation
+          //
+          routerDelegates[currentIndex]
+              .beamToNamed(index == 0 ? '/books' : '/articles');
+
+          // OPTION 2:
+          // Advantage: keeps details pages
+          // Disadvantage: does not update URL
+          //
+          /*routerDelegates[currentIndex].parent?.updateRouteInformation(
```

Comment:
We'll fix this by adding your `setApplicationSwitcherDescription` into `updateRouteInformation` also. We missed that.
The crux here is that `updateRouteInformation` will just update the URL bar in browser, without calling `build` (where `setApplicationSwitcherDescription` is)

Reply:
Great 👍😀